### PR TITLE
add component (both js and hbs) Add File choice

### PR DESCRIPTION
**Pull request by kagemusha**
Takes care of issue #35.

One difference from discussions there is that if user enters file extension, get snarky error msg instead of stripping it off :-), but this can be changed if the latter felt better.

Tests to come.

### CODE REVIEW

File: app/gist/controller.js [hunk 1]
```diff
@@ -86,6 +86,46 @@ export default Ember.Controller.extend({
...
+  },
+  /*
+   *  Test path is valid.  Presently only tests components are hyphenated.
+   */
+  hasInvalidPath(type, path){
+    let errorMsg = null;
+    if (type === 'component'){
+      let file = path.split('/').slice(-1)[0];
+      if (file.indexOf('.') > -1) {
+        errorMsg = "What part of WITHOUT FILE EXTENSION did you not understand ;-) ?";
```
Comment (rwjblue) [inline app/gist/controller.js]: LOL

Comment (rwjblue) [inline app/gist/controller.js]: Might want to make it *slightly* kinder :smiling_imp:

**Comment by Gaurav0** [comment]
You have a jshint error, and you have some conflicts.

File: app/gist/controller.js (hunk 1 above)
Comment (Gaurav0) [inline app/gist/controller.js]: @rwjblue @joostdevries Do we want an error message at all or do we want to automatically strip the extension?

Comment (kagemusha) [inline app/gist/controller.js]: @Gaurav0 @rwjblue @joostdevries imagine most of the time people will want to do what stripping accomplishes, so despite my affection for the snarky msg, gonna strip the extension if no objections

Comment (Gaurav0) [inline app/gist/controller.js]: Sounds good.

**Comment by Gaurav0** [comment]
@kagemusha This looks great! Can you do the following:

1. When adding a component, it should open the js file in column 1 and the template in column 2
2. Place the two legacy component options (for individual hbs and js) in a subdropdown (subflyout) menu.

**Comment by kagemusha** [comment]
@Gaurav0 added.  pls take a look.

**Comment by Gaurav0** [comment]
:+1:

**Pull request merged**